app/code/Magento/Catalog/Model/Product/Image/Cache.php

```diff
@@ -127,4 +154,97 @@ protected function processImageData(Product $product, array $imageData, $file)
 
         return $this;
     }
+
+    /**
+     * Get themes in use
+     *
+     * @return array
+     */
+    private function getThemesInUse(): array
+    {
+        $themesInUse = [];
+
+        $registeredThemes = $this->themeCollection->loadRegisteredThemes();
+        $storesByThemes   = $this->themeCustomizationConfig->getStoresByThemes();
+
+        foreach ($registeredThemes as $registeredTheme) {
+            if (array_key_exists($registeredTheme->getThemeId(), $storesByThemes)) {
+                $themesInUse[] = $registeredTheme;
+            }
+        }
+
+        $connection = $this->attribute->getResource()->getConnection();
+
+        $productEntityTypeId = \Magento\Catalog\Setup\CategorySetup::CATALOG_PRODUCT_ENTITY_TYPE_ID;
+
+        $productCustomDesignAttributeId = $this->attribute->loadByCode($productEntityTypeId, 'custom_design')->getId();
+
+        $productSql = $connection
+            ->select()
+            ->from(
+                ['eav' => $connection->getTableName('catalog_product_entity_varchar')],
+                ['value']
+            )
+            ->where('eav.attribute_id = ?', $productCustomDesignAttributeId)
+            ->where('eav.value > 0')
+            ->group('value');
+
+        $productThemeIds = $connection->fetchCol($productSql);
+
+        if (count($productThemeIds)) {
+            foreach ($productThemeIds as $productThemeId) {
+                if (array_key_exists($productThemeId, $storesByThemes)
+                    && !array_key_exists($productThemeId, $themesInUse) ) {
+                    $themesInUse[] = $this->themeCollection->load($productThemeId);
+                }
+            }
+        }
+
+        $categoryEntityTypeId = \Magento\Catalog\Setup\CategorySetup::CATEGORY_ENTITY_TYPE_ID;
+
+        $categoryCustomDesignAttributeId = $this->attribute->loadByCode($categoryEntityTypeId, 'custom_design')->getId();
```

Comment:
Please use \Magento\Eav\Model\Config::getAttribute instead of load entity by model